Feature/pagination scroll

SDBowen opened the pull request:
This is a continuation of the work done in PR #68 by @italo-aurelio. 

It adds a smooth scroll effect when the user selects the next page, and the browser returns to the top of the paginated list.

Test in these workspaces:

[workspace](https://sdb--arcaplanet.myvtex.com/blog/category/consigli?page=1)

[workspace](https://sdb--arteni.myvtex.com/blog/category/moda?__bindingAddress=www.arteni.it/it&page=1)

polishq reviewed (commented):
Two things:

- Scroll should not occur on the initial page load; currently, when you first load the second Workspace link, the page automatically scrolls down to the top of the blog list
- What's the rationale of adding `window.pageYOffset` to the bounding box? This seems to result in the scroll ending a little off target.

SDBowen commented:
> Two things:
> 
> * Scroll should not occur on the initial page load; currently, when you first load the second Workspace link, the page automatically scrolls down to the top of the blog list
> * What's the rationale of adding `window.pageYOffset` to the bounding box? This seems to result in the scroll ending a little off target.

Ah, yeah, added a variable to skip the behavior on the initial page load.


For that off target amount. It comes from the page navigation header.

`getBoundingClientRect().top` gives us the distance of the top of the current view to the top of the pagination element. (when the view is below the element, it is a negative number)

`window.pageYOffset` gives us the distance from the top of the page to the top of the current view.

Together we get the distance from the top of the page, to the top of the pagination component. This results in the navigation header covering part of the pagination element.

How about adding some fixed space? I have updated the workspaces in the PR.
